### PR TITLE
fix(20.04): fix arch specific paths for openjdk-8

### PR DESCRIPTION
**Pull request by cjdcordeiro**
Same as #149 

@vpa1977 can you also pls double check this one?

### CODE REVIEW

**Comment by vpa1977** [comment]
/client and /server hotspot machines could be built together and present in the same installation, though it is not something that openjdk-8 packaging does. 
Maybe would it be possible to split those into separate slices (if chisel allows) ? 
Otherwise the change look fine.

**Comment by cjdcordeiro** [comment]
> /client and /server hotspot machines could be built together and present in the same installation, though it is not something that openjdk-8 packaging does. Maybe would it be possible to split those into separate slices (if chisel allows) ? Otherwise the change look fine.

when you say 

> not something that openjdk-8 packaging does

can you please clarify? for `armhf` it seems that there's only `client`, while `server` is only available for other archs. Are you saying that, for `armhf`, the `client` should not be installed as part of the current slice?

**Comment by vpa1977** [comment]
> can you please clarify? for `armhf` it seems that there's only `client`, while `server` is only available for other archs. Are you saying that, for `armhf`, the `client` should not be installed as part of the current slice?

I apologise for confusion. Yes, we build client for the armhf. 
I was wondering if we can do something like:

```
slices:
  # The "core" slice provides a minimal, yet functional JRE8.
  core:
    essential:
      - libc6_libs
      - libgcc-s1_libs
      - libstdc++6_libs
      - zlib1g_libs
      - openjdk-8-jre-headless_client_vm
      - openjdk-8-jre-headless_server_vm

... 
```
and move the client and server vm contents into separate slices.

File: slices/openjdk-8-jre-headless.yaml [hunk 1]
```diff
@@ -27,8 +27,10 @@ slices:
       /usr/lib/jvm/java-8-openjdk-*/jre/lib/*/libverify.so:
       /usr/lib/jvm/java-8-openjdk-*/jre/lib/*/libzip.so:
       /usr/lib/jvm/java-8-openjdk-*/jre/lib/*/libjsig.so:
-      /usr/lib/jvm/java-8-openjdk-*/jre/lib/*/server/libjsig.so:
-      /usr/lib/jvm/java-8-openjdk-*/jre/lib/*/server/libjvm.so:
+      /usr/lib/jvm/java-8-openjdk-*/jre/lib/*/client/libjsig.so: {arch: armhf}
+      /usr/lib/jvm/java-8-openjdk-*/jre/lib/*/client/libjvm.so: {arch: armhf}
+      /usr/lib/jvm/java-8-openjdk-*/jre/lib/*/server/libjsig.so: {arch: [amd64,arm64,ppc64el]}
+      /usr/lib/jvm/java-8-openjdk-*/jre/lib/*/server/libjvm.so: {arch: [amd64,arm64,ppc64el]}
```
Comment (rebornplusplus) [inline slices/openjdk-8-jre-headless.yaml:33]: nit: https://github.com/canonical/chisel-releases/pull/149#discussion_r1503688416.

**Review by rebornplusplus — APPROVED** [review]
Looks good to me. I will let you guys sort out the organizing of files.

File: slices/openjdk-8-jre-headless.yaml (hunk 1 above)
Comment (cjdcordeiro) [reply slices/openjdk-8-jre-headless.yaml]: done

**Comment by cjdcordeiro** [comment]
> > can you please clarify? for `armhf` it seems that there's only `client`, while `server` is only available for other archs. Are you saying that, for `armhf`, the `client` should not be installed as part of the current slice?
> 
> I apologise for confusion. Yes, we build client for the armhf. I was wondering if we can do something like:
> 
> ```
> slices:
>   # The "core" slice provides a minimal, yet functional JRE8.
>   core:
>     essential:
>       - libc6_libs
>       - libgcc-s1_libs
>       - libstdc++6_libs
>       - zlib1g_libs
>       - openjdk-8-jre-headless_client_vm
>       - openjdk-8-jre-headless_server_vm
> 
> ... 
> ```
> 
> and move the client and server vm contents into separate slices.

Ah...I see. We could do that indeed. It would be a refactoring task that I would address separately from this PR though, cause it would mean doing it for other releases too. See https://github.com/canonical/chisel-releases/issues/152

**Review by linostar — APPROVED** [review]
LGTM.

**Pull request merged**